clang/lib/Format/UnwrappedLineFormatter.cpp

```diff
@@ -686,8 +686,8 @@ class LineJoiner {
     }
     Limit = limitConsideringMacros(I + 1, E, Limit);
     AnnotatedLine &Line = **I;
-    if (Line.First->isNot(tok::kw_do) && Line.First->isNot(tok::kw_else) &&
-        Line.Last->isNot(tok::kw_else) && Line.Last->isNot(tok::r_paren)) {
+    if (Line.First->isNotOneOf(tok::kw_do, tok::kw_else) &&
```

Comment:
I'm wondering if we should have  Line.beginsWith(),  Line.beingWithOneOf() and Line.doesNotBeingWithOneOf() I get a bit blind after a while in this code...